hpy/debug/src/stacktrace.c

```diff
@@ -0,0 +1,93 @@
+#include "debug_internal.h"
+
+#if (__linux__ && __GNU_LIBRARY__)
+
+// Basic implementation that uses backtrace from glibc
+
+#include <stdlib.h>
+#include <string.h>
+#include <execinfo.h>
+
+static inline int max_s(size_t a, size_t b) {
+    return a > b ? a : b;
+}
+
+void create_stacktrace(char **target, HPy_ssize_t max_frames_count) {
```

Comment:
I suppose this is fine for now but also probably very slow and memory inefficient because it creates a textual representation of the stack trace for every handle.
I *think* it's possible to save only the addresses here, and delay the calls to `backtrace_symbols` and the construction of the textual repr only if/when it's really needed.

Reply:
Good point. I wanted to make it generic such that we can easily swap the backend. Thinking about it now, we may also have something like this:
```
void *stracktrace_create(HPy_ssize_t max_frames_count);
void stacktrace_print(void *stacktrace, char **target, HPy_ssize_t max_frames_count);
void stacktrace_free(void *stacktrace);
```
but I'll leave that for future work.

Reply:
yes, +1 for both your suggestion and for doing it in the future